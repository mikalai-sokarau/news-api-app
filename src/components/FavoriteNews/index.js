import React from "react";
import NewSingle from "../../components/SingleNews";
import StyledMessage from "./style";

const FavoriteNews = props => {
  const renderNews = () =>
    props.news.map(item => (
      <NewSingle
        key={item.data.url}
        item={item.data}
        removeNewsFromFavorite={props.removeNewsFromFavorite}
        checked
      />
    ));

  const emptyFavoriteNewsMessage = (
    <StyledMessage>
      <span>There are no favorite news yet. First add a few from</span>
      <a onClick={() => props.history.goBack()} className="waves-effect waves-light btn-small">
        Home
      </a>
    </StyledMessage>
  );

  return (
    <div className="row">
      {props.news.length ? renderNews() : emptyFavoriteNewsMessage}
    </div>
  );
};

export default FavoriteNews;
