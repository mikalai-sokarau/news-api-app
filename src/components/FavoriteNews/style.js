import styled from "styled-components";

const StyledMessage = styled.div`
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  position: absolute;
  top: 50%;

  span {
    font-size: 1.3rem;
    margin-right: 10px;
    font-family: KaushanScript;
  }
`;

export default StyledMessage;
