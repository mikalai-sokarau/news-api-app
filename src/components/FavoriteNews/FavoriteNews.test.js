import React from 'react';
import { shallow } from 'enzyme';
import FavoriteNews from './index';

describe('FavoriteNews', () => {
  test('should render a Message component if news array is empty', () => {
    const wrapper = shallow(
      <FavoriteNews
        news={[]}
        buttonName="buttonName"
        messageText="messageText"
        buttonClickHandler={() => {}}
      />
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('Message')).toHaveLength(1);
    expect(wrapper.find('SingleNews')).toHaveLength(0);
  });

  test('should render news, if they are exist', () => {
    const fakeNews = [
      { data: { url: '/fake1' } },
      { data: { url: '/fake2' } },
      { data: { url: '/fake3' } }
    ];
    const wrapper = shallow(<FavoriteNews news={fakeNews} iconClickHandler={() => {}} />);

    expect(wrapper.find('Message')).toHaveLength(0);
    expect(wrapper.find('SingleNews')).toHaveLength(3);
  });
});
