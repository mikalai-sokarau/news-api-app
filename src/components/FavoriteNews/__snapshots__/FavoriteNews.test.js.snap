// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteNews should render a Message component if news array is empty 1`] = `
<div
  className="row"
>
  <Message
    buttonClickHandler={[Function]}
    buttonName="buttonName"
    text="messageText"
  />
</div>
`;
